vtkext/private/module/vtkF3DRenderer.cxx

```diff
@@ -1560,6 +1651,108 @@ void vtkF3DRenderer::ResetCameraClippingRange()
+{
+  const auto isMatrixAxisAligned = [](const vtkMatrix4x4* m, const double tol = 1e-8)
+  {
+    for (size_t i = 0; i < 3; ++i)
+    {
+      size_t nonzerosI = 0;
+      size_t nonzerosJ = 0;
+      for (size_t j = 0; j < 3; ++j)
+      {
+        if (::abs(m->Element[i][j]) > tol)
+          nonzerosI++;
+        if (::abs(m->Element[j][i]) > tol)
+          nonzerosJ++;
+      }
+      if (nonzerosI > 1 || nonzerosJ > 1)
+        return false;
+    }
+    return true;
+  };
+
+  /* Use `PokeMatrix` around the call to `GetBounds()` to extend box.
+   * Only gives the thightest bounds if the transformation is axis-aligned. */
+  const auto extendBoxAxisAligned = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkNew<vtkMatrix4x4> tmpMatrix;
+    vtkMatrix4x4::Multiply4x4(matrix, prop3d->GetMatrix(), tmpMatrix);
+    prop3d->PokeMatrix(tmpMatrix);
+
+    box.AddBounds(prop3d->GetBounds());
+
+    prop3d->PokeMatrix(NULL);
+  };
+
+  /* Use custom logic to extend box.
+   * Should give the tightest bounds even when non-axis-aligned */
+  const auto extendBoxArbitrary = [&](vtkProp3D* prop3d, vtkBoundingBox& box)
+  {
+    vtkActor* actor = dynamic_cast<vtkActor*>(prop3d);
+    if (actor)
+    {
+      vtkPolyDataMapper* polyMapper = dynamic_cast<vtkPolyDataMapper*>(actor->GetMapper());
+      if (polyMapper)
+      {
+        vtkNew<vtkTransform> t;
+        t->Concatenate(*matrix->Element);
+        t->Concatenate(actor->GetMatrix());
+        vtkPolyData* polydata = polyMapper->GetInput();
+        if (polydata)
+        {
+          for (vtkIdType i = 0; i < polydata->GetNumberOfPoints(); ++i)
+          {
+            box.AddPoint(t->TransformPoint(polydata->GetPoint(i)));
+          }
+          return true;
+        }
+      }
+    }
+    return false;
+  };
+
+  const bool isAxisAligned = isMatrixAxisAligned(matrix);
+  vtkBoundingBox box;
+
+  /* use `ComputeVisiblePropBounds()`'s logic to iterate `vtkProp3D`s contributing to the bounds */
+  {
+    vtkProp* prop;
+    vtkCollectionSimpleIterator pit;
+    for (this->Props->InitTraversal(pit); (prop = this->Props->GetNextProp(pit));)
+    {
+      if (prop->GetVisibility() && prop->GetUseBounds())
+      {
+        const double* bounds = prop->GetBounds();
+        if (bounds != nullptr && vtkMath::AreBoundsInitialized(bounds))
+        {
+          vtkProp3D* prop3d = dynamic_cast<vtkProp3D*>(prop);
+          if (prop3d)
+          {
+            if (isAxisAligned)
+            {
+              extendBoxAxisAligned(prop3d, box);
+            }
+            else
+            {
+              if (!extendBoxArbitrary(prop3d, box))
+              {
+                const std::string repr = std::string(prop3d->GetClassName());
+                F3DLog::Print(F3DLog::Severity::Warning,
+                  "Could not properly account for " + repr +
+                    " in non-axis-aligned bounds computation");
+                extendBoxAxisAligned(prop3d, box);
```

Comment:
This is a fallback in the unlikely case there is no custom bounds implementation for a given `prop3d`. It should not happen if we have all the possible props/actors/mappers types handled so I don't think it can be covered by tests.

Reply:
then let's assert?

Reply:
couldn't a plugin use some exotic prop we haven't covered tho?